### ISSUE TITLE
[BUG]: boilerplate jest is broken

### ISSUE THREAD

**Issue by davidnixon** [IBMa/equal-access#1586]
### Project

other

### Browser

Chrome

### Operating system

Other

### Description

  ● Test suite failed to run

    Jest encountered an unexpected token

    /data1/projects/equal-access/accessibility-checker/boilerplates/jest/node_modules/axios/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import axios from './lib/axios.js';
                                                                                             ^^^^^^

    SyntaxError: Cannot use import statement outside a module

       8 | 'use strict';
       9 |
    > 10 | const aChecker = require("accessibility-checker");
         |                  ^
      11 |
      12 | async function toBeAccessible(node, label) {
      13 |   let results = await aChecker.getCompliance(node, label);

### Workaround 1 - change axios
```sh
npm install axios@^0.27.2
rsync -a node_modules/axios/ node_modules/accessibility-checker/node_modules/axios/
```
> Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 passed, 2 total
Snapshots:   0 total
Time:        2.06 s, estimated 4 s
Ran all test suites matching /test\/basic.test.js/i.

### Workaround 2 - change import
There are 2 imports for axios. Patch the code in `node_modules/accessibility-checker/lib/ACEngineManager.js` and `node_modules/accessibility-checker/lib/common/api-ext/Fetch.js` to use `require("axios/dist/browser/axios.cjs")`

That fixes the jest test but breaks the command line tool

### More info

- running from the command line works fine `npx accessibility-checker some.html`

### Steps to reproduce

clone repo
nvm use 16
```sh
cd equal-access/accessibility-checker/boilerplates/jest
npm install
npm run test
```

**Comment by tombrunet** [comment]
Thank you. Looks like something changed in axios, we'll take a look. For node 18, it's supposed to automatically flip to `fetch` and not use axios at all, but looks like axios is still getting imported anyway.

**Comment by tombrunet** [comment]
@davidnixon It looks like if you're using modules, we have to use axios 1.0.0+ (which we are). If you're not, we would have to use 0.x.x. The better fix on our end is to use `fetch` instead, which is available by default in Node 18 and by command-line flag in Node 16. We use fetch for some of the code, but I found one place that we still had axios even when fetch was available that I need to fix. Any reason that you wouldn't be able to update to Node 18?

Longer term, we're running into more and more dependencies that simply don't support the cjs / require syntax and are moving more to the ejs module / import syntax.

**Comment by tombrunet** [comment]
@davidnixon So there are a few things going on here that I'll list below. To get you up and running with what's out there now, you need to delete the "jest" section from package.json and then have this in your jest.config.js:
```js
module.exports = {
    "moduleNameMapper": {
        // Force module uuid to resolve with the CJS entry point, because Jest does not support package.json.exports. See https://github.com/uuidjs/uuid/issues/451
        // Note that this comes through the exceljs module
        "uuid": require.resolve('uuid'),
        "axios": require.resolve('axios')
    },
    "setupFilesAfterEnv": [
        "./setupAfterEnv.js"
    ],
    "transformIgnorePatterns": [
        "ace-node\\.js"
    ]
}
```

So, some things going on:
* Even when running Node 18, **jest 26.x** does not support `fetch`. **jest 29.x** does. So, when you get to Node 18 and **jest 29.x** and #1596 is in release, the **axios** dependency should go away so that line of the `moduleNameMapper` can be removed.
* In #1596 we remove our dependency on the **uuid** package and use **crypto** directly, but the **exceljs** package still needs to update to **uuid 9+** before you can remove the **uuid** line from the `moduleNameMapper`.

I'm going to close this since the above change works with the deployed release and the example code is in the PR. Will continue with the PR to make this cleaner and will have to monitor exceljs until the moduleNameMapper is no longer needed.

**Comment by davidnixon** [comment]
@tombrunet The above fix is working for me. I do get an **enormous** amount of console.error output from my test though. This is repeated about 40 times in my test. 

> Error: Not implemented: window.computedStyle(elt, pseudoElt) ... + 1000 more character

I added this to `toBeAccessible` to work around the issue.
```js
async function toBeAccessible(node, label) {
  console.error = () => {};
  let results = await aChecker.getCompliance(node, label);
...
```
